src/compiler/commandLineParser.ts

```diff
@@ -996,9 +996,9 @@ namespace ts {
         return { options, errors };
     }
 
-    export function convertTypingOptionsFromJson(jsonOptions: any, basePath: string, configFileName?: string): { options: TypingOptions, errors: Diagnostic[] } {
+    export function convertTypeAcquisitionFromJson(jsonOptions: any, basePath: string, configFileName?: string): { options: TypeAcquisition, errors: Diagnostic[] } {

```

Comment:
Adding support for the deprecated API (typingOptions.enableAutoDiscovery) just in case